vcdriver/vm.py

```diff
@@ -342,6 +346,73 @@ def winrm(self, script, winrm_kwargs=dict(), **kwargs):
             else:
                 return status, stdout, stderr
 
+    @configurable([
+        ('Virtual Machine Remote Management', 'vcdriver_vm_winrm_username'),
+        ('Virtual Machine Remote Management', 'vcdriver_vm_winrm_password')
+    ])
+    def winrm_upload(
+            self, remote_path, local_path, step=1024, winrm_kwargs=dict(),
+            **kwargs
+    ):
+        """
+        Copy a file through winrm
+        :param remote_path: The remote location
+        :param local_path: The local local
+        :param step: Number of bytes to send in each chunk
+        :param winrm_kwargs: The pywinrm Protocol class kwargs
+
+        :return: A tuple with the status code, the stdout and the stderr
+        """
+        if self._vm_object:
+            winrm_session = winrm.Session(
+                target=self.ip(),
+                auth=(
+                    kwargs['vcdriver_vm_winrm_username'],
+                    kwargs['vcdriver_vm_winrm_password'],
+                ),
+                read_timeout_sec=self.timeout + 1,
+                operation_timeout_sec=self.timeout,
+                **winrm_kwargs
+            )
+            with open(local_path, 'rb') as f:
+                contents = f.read()
```

Comment:
Why read the whole file in advance rather than reading it in chunks?

Reply:
That's a good point, I can fix that.